providers/interface.go

```diff
@@ -4,7 +4,26 @@ must conform to.
 */
 package providers
 
-import "github.com/hashicorp/hcl/v2"
+import (
+	"sync"
+
+	"github.com/hashicorp/hcl/v2"
+)
+
+var (
+	FactoryMap Factories
+	once       sync.Once
+)
+
+// Register registers a provider factory.
+func Register(id string, f Factory) {
+	if FactoryMap == nil {
+		FactoryMap = make(map[string]Factory)
+	}
+	once.Do(func() {
+		FactoryMap[id] = f
```

Comment:
Is the `once` instance not shared between providers here? So only the first provider to init will register?

Or does each provider package have a unique `once` because they all import separately? (But then, a single package can't register multiple providers?)

Reply:
Yeah, that was a bad call on my part to use `sync.Once` and a good catch on yours.